### DISCUSSION TITLE
wich stack to use in onder to get an interactive map ?

### DISCUSSION THREAD

**Discussion by bioinfornatics** [JanusGraph/janusgraph#2674] [General]
Dear,

I would like to use JanusGraph to visualize a multi layers map (a la google map/open street map):  satellite, country, state , road
Did you have any pointer to a library to get a such dynamic rendering (zoom in/out) ?

Thanks

best regards

**Comment by mbrukman** [comment]
I haven't seen JanusGraph used for this; OpenStreetMap typically does this by serving map tiles via [MapServer](https://wiki.openstreetmap.org/wiki/MapServer) (there's no graph data in the imagery; image tiles are keyed by [tile coordinates](https://leafletjs.com/reference-1.7.1.html#tilelayer)) and a client-side library such as [LeafletJS](https://leafletjs.com/); there are many alternatives for both tile servers and client-side libraries.

You can also use JS libraries without a tileserver if you use hosted tile server APIs, which may be free or paid services.

**Reply by bioinfornatics** [reply]
thanks @mbrukman for your insight
